cmd/subnetcmd/elastic.go

```diff
@@ -87,39 +167,59 @@ func transformElasticSubnet(_ *cobra.Command, args []string) error {
 	}
 
 	var network models.Network
-	if transformLocal {
+	switch {
+	case deployTestnet:
+		network = models.Fuji
+	case deployMainnet:
+		network = models.Mainnet
+	case transformLocal:
 		network = models.Local
 	}
 
 	if network == models.Undefined {
-		networkToUpgrade, err := promptNetworkElastic(sc, "Which network should transform into an elastic Subnet?")
+		networkToUpgrade, err := selectNetworkToTransform(sc)
 		if err != nil {
 			return err
 		}
 		switch networkToUpgrade {
+		case localDeployment:
+			network = models.Local
 		case fujiDeployment:
-			return errors.New("elastic subnet transformation is not yet supported on Fuji network")
-		case mainnetDeployment:
+			network = models.Fuji
+		default:
 			return errors.New("elastic subnet transformation is not yet supported on Mainnet")
 		}
 	}
 
-	if checkIfSubnetIsElasticOnLocal(sc) {
-		return fmt.Errorf("%s is already an elastic subnet", subnetName)
+	if outputTxPath != "" {
+		if _, err := os.Stat(outputTxPath); err == nil {
+			return fmt.Errorf("outputTxPath %q already exists", outputTxPath)
+		}
 	}
 
-	subnetID := sc.Networks[models.Local.String()].SubnetID
+	if len(ledgerAddresses) > 0 {
+		useLedger = true
+	}
+
+	if useLedger && keyName != "" {
+		return ErrMutuallyExlusiveKeyLedger
+	}
+
+	subnetID := sc.Networks[network.String()].SubnetID
+	if os.Getenv(constants.SimulatePublicNetwork) != "" {
```

Comment:
can you move the following check later, so the check for this os env is made once?